src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+    private final int indexMethod = stackTraceElements.length - 2;
+    private StackTraceElement element = stackTraceElements[indexMethod];
+    private Object[] elementData;
+    private int size;
+
+    ArrayList() {
+        size = 0;
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size < index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't add element by this index, method: " + element);
+        }
+        if (++size >= elementData.length) {
+            grow();
+        }
+        if (index == 0) {
+            System.arraycopy(elementData, 0, elementData, 1, size);
+        } else {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        }
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int currentCapacity = elementData.length;
+        while (currentCapacity + list.size() > elementData.length) {
+            grow();
+        }
+        for (int j = 0; j < list.size(); j++) {
+            add(list.get(j));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        T removedElement = (T)elementData[index];
+        if (index == 0) {
+            System.arraycopy(elementData, 1, elementData, 0, --size);
+        } else {
+            System.arraycopy(elementData, index + 1, elementData, index, --size - index);
+        }
+        elementData[size] = null;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = getIndexOfElement(element);
+        if (index == -1) {
+            throw new NoSuchElementException(
+                    "There is no such element in ArrayList, method: " + element);
+        }
```

Comment:
the same here you check the index in method `remove --> indexCheck ` so it’s  code redundant